content/en/docs/reference/using-api/api-concepts.md

```diff
@@ -331,6 +331,35 @@ For example:
 Accept: application/vnd.kubernetes.protobuf, application/json
 ```
 
+### CBOR resource encoding {#cbor-encoding}
+
+{{< feature-state feature_gate_name="CBORServingAndStorage" >}}
+
+With the `CBORServingAndStorage` [feature
+gate](/docs/reference/command-line-tools-reference/feature-gates/) enabled, request and response
+bodies for all built-in resource types and all resources defined by a {{< glossary_tooltip
+term_id="CustomResourceDefinition" text="CustomResourceDefinition" >}} may be encoded to the
+[CBOR](https://www.rfc-editor.org/rfc/rfc8949) binary data format. CBOR is also be supported at the
+{{< glossary_tooltip text="aggregation layer" term_id="aggregation-layer" >}} if it is enabled in
+individual aggregated API servers.
+
+Clients should indicate the IANA media type `application/cbor` in the `Content-Type` HTTP request
+header when the request body contains a CBOR-encoded object, and in the `Accept` HTTP request header
+when prepared to accept a CBOR-encoded object in the response. API servers will use
+`application/cbor` in the `Content-Type` HTTP response header when the response body contains a
+CBOR-encoded object.
```

Comment:
```suggestion
Clients should indicate the IANA media type `application/cbor` in the `Content-Type` HTTP request
header when the request body contains a CBOR-encoded resource, and in the `Accept` HTTP request header
when prepared to accept a CBOR-encoded resource in the response. API servers will use
`application/cbor` in the `Content-Type` HTTP response header when the response body contains a
CBOR-encoded resource representation.
```
I assume CBOR works even if the thing you interact with is not an object.

Reply:
I don't think that is accurate. For example, a client can make a request to delete the resource "/api/v1/namespaces/foo/pods/bar" and the request body will not contain a representation of any resource, it will contain a serialized DeleteOptions object. Similarly, an update to the subresource "/api/v1/namespaces/foo/pods/bar/status" would contain a Pod object, and updates to "/api/apps/v1/namespaces/foo/replicasets/bar/scale" would encode Scale objects in the request and response bodies.

The Serializer interface itself only accepts and returns objects. If the content type of a request or response is `application/cbor`, then its body is always a serialized object. The only exceptions (apply configurations and sequences of objects) use different content types.

Reply:
I don't think the body of a CBOR-encoding request or response is always
- a CBOR object (CBOR's equivalent of a JSON object is formally a _map_)
- a Kubernetes object (see previous feedback)
- a Go object (we don't know what implementation language was used;  it could be an aggregated API server)

Try to avoid using the confusing term "object" here. Feel free to propose a different rewording.

Reply:
It is always a Kubernetes object, that's the usage I was intending here.

Reply:
Ah, that's not the case. For example a delete options request body is not Kubernetes [object](https://kubernetes.io/docs/concepts/overview/working-with-objects/).
Some more things that are not Kubernetes objects but that you might read / write in CBOR:
- a TokenReview
- an APIGroupList (no `.metadata`, no `.spec`, no `.status` so not an object)
- arguably, the response from reading a `scale` subresource (becayse it is a subresource of whatever object it represents the scale of, not an object in its own right)

Reply:
> Ah, that's not the case. For example a delete options request body is not Kubernetes [object](https://kubernetes.io/docs/concepts/overview/working-with-objects/).
Some more things that are not Kubernetes objects but that you might read / write in CBOR:

I don't agree with the... separately from this PR, the docs at that link should be rephrased... objects are serialized types with apiVersion / kind.

* Objects are not *always* persisted in etcd (though they usually are) – DeleteOptions serialized into the body of a request, Status objects returned in a response, TokenReview / SubjectAccessReview / TokenRequest objects serialized into requests, and configuration objects read from config files are all objects
* Objects are not *always* used for workload and reconciliation patterns, though they often are

Reply:
To avoid the concern: reword and avoid the term "object".

Reply:
Thanks for this discussion. I see that this definition of the term "object" is widespread in the existing user-facing documentation, even though it differs from the meaning that is normally understood when talking about API machinery. I've replaced the term here with the less-specific but still correct "CBOR-encoded data item" (https://www.rfc-editor.org/rfc/rfc8949.html#section-1.2-4.2) to sidestep the issue.